matching-notification-api/src/main/java/org/phenotips/matchingnotification/storage/internal/DefaultMatchStorageManager.java

```diff
@@ -502,6 +503,34 @@ private void copyToHistory(Session session, Collection<PatientMatch> matches)
         }
     }
 
+    /**
+     * Load {@code match} with specified parameters.
+     *
+     * @param referencePatientId id of reference patient
+     * @param referenceServerId id of the server that hosts reference patient
+     * @param matchedPatientId id of matched patient
+     * @param matchedServerId id of the server that hosts matched patient
+     * @return match
+     */
+    private PatientMatch loadMatchBetweenPatients(String referencePatientId, String referenceServerId,
+        String matchedPatientId, String matchedServerId)
+    {
+        if (StringUtils.isNotEmpty(referencePatientId)) {
+            // referencePatientId is "reference patient"
+            Criterion directMatch = Restrictions.and(
+                this.patientIsReference(referencePatientId, referenceServerId),
+                this.patientIsMatch(matchedPatientId, matchedServerId));
+            List<PatientMatch> matches = this.loadMatchesByCriteria(directMatch);
+            if (matches.isEmpty()) {
```

Comment:
it is better (less `returns` and no `else`) to
```
if (!matches.isEmpty()) {
  return matches.get(0);
}
```
then `return null;` outside all `if`-s